src/Hosting/Hosting/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -73,5 +86,59 @@ private bool StartsWithBasePath(string subpath, out PathString rest)
         {
             return new PathString(subpath).StartsWithSegments(BasePath, FilePathComparison, out rest);
         }
+
+        private class FakeRoot : IDirectoryContents
```

Comment:
I would call this something like StaticWebAssetsDirectortRoot or similar. Avoid using terms like Fake as that sounds like its part of a test.